### PR TITLE
Added postgresql 10 + postgis 2.4.0

### PR DESCRIPTION
**Pull request by tdamsma**
Added postgresql 10 + postgis 2.4.0, and postgresql 9.6 + postgis 2.4.0.

To get the postgis 2.3 builds passing, I had to remove line 11 ` postgis=$POSTGIS_VERSION \` from the template, as postgis 2.3 has been purged from the apt repo. I don't understand why postgis needs to be installed after postgresql-x-postgis-x is installed before, everything seem to just work without it. But as I don't understand why it was there, I can't really judge if it is a bad idea to remove it. 

As postgresl-10 is `FROM debian:stretch`, I modified the update script to take that into account

### CODE REVIEW

**Comment by tdamsma** [comment]
Apparently this is not as easy as expected, will look into it tomorrow. I can get the builds passing with postgis 2.4, but not yet with 2.3

File: Dockerfile.template [hunk 1]
```diff
@@ -8,7 +8,6 @@ RUN apt-get update \
       && apt-get install -y --no-install-recommends \
            postgresql-$PG_MAJOR-postgis-$POSTGIS_MAJOR=$POSTGIS_VERSION \
            postgresql-$PG_MAJOR-postgis-$POSTGIS_MAJOR-scripts=$POSTGIS_VERSION \
-           postgis=$POSTGIS_VERSION \
```
Comment (md5) [inline Dockerfile.template]: Can we discuss this removal? I know they don't version this package name, so is the issue that only the 2.4 version is now available? If so, then I'd like to remove 2.3 entirely.

**Comment by md5** [comment]
Can you squash your commits? I think it would make sense to see a single commit for the 2.3 package upgrades (assuming we don't remove 2.3), another commit for adding the 2.4 images, and another commit for the updates to `update.sh`.

File: Dockerfile.template (hunk 1 above)
Comment (md5) [reply Dockerfile.template]: I just saw that you commented on this already.

The `postgis` package provides a number of PostGIS binaries if I recall. There is some discussion in https://github.com/appropriate/docker-postgis/issues/8 and https://github.com/appropriate/docker-postgis/issues/26

**Comment by tdamsma** [comment]
That would indeed make perfect sense, but my git-fu is lacking a bit. You could also just squash-merge the whole bunch, as the updates are a bit related: I didn't want to update the 2.3 images, but I had to cause they used outdated packages. Anyways, I'll give it a shot

**Comment by tdamsma** [comment]
Resubmitted as https://github.com/appropriate/docker-postgis/pull/64

**Pull request closed without merging**